myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/TaskFactory.java

```diff
@@ -125,82 +109,40 @@ private String getConfigurationUrl() {
       }
     }
 
-    private CommandInfo getCommandInfo() {
+    private CommandInfo getCommandInfo(NMProfile profile, NMPorts ports) {
       MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
       CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
-      if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
-                /*
-                 TODO(darinj): Overall this is messier than I'd like. We can't let mesos untar the distribution, since
-                 it will change the permissions.  Instead we simply download the tarball and execute tar -xvpf. We also
-                 pull the config from the resource manager and put them in the conf dir.  This is also why we need
-                 frameworkSuperUser. This will be refactored after Mesos-1790 is resolved.
-                */
+      ExecutorCommandLineGenerator clGenerator;
+      String cmd;
 
+      if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
         //Both FrameworkUser and FrameworkSuperuser to get all of the directory permissions correct.
         if (!(cfg.getFrameworkUser().isPresent() && cfg.getFrameworkSuperUser().isPresent())) {
           throw new RuntimeException("Trying to use remote distribution, but frameworkUser" +
-              "and/or frameworkSuperUser not set!");
+            "and/or frameworkSuperUser not set!");
         }
-
-        LOGGER.info("Using remote distribution");
-
-        String nmURIString = myriadExecutorConfiguration.getNodeManagerUri().get();
-
-        //TODO(DarinJ) support other compression, as this is a temp fix for Mesos 1760 may not get to it.
-        //Extract tarball keeping permissions, necessary to keep HADOOP_HOME/bin/container-executor suidbit set.
-        String tarCmd = "sudo tar -zxpf " + getFileName(nmURIString);
-
-        //We need the current directory to be writable by frameworkUser for capsuleExecutor to create directories.
-        //Best to simply give owenership to the user running the executor but we don't want to use -R as this
-        //will silently remove the suid bit on container executor.
-        String chownCmd = "sudo chown " + cfg.getFrameworkUser().get() + " .";
-
-        //Place the hadoop config where in the HADOOP_CONF_DIR where it will be read by the NodeManager
-        //The url for the resource manager config is: http(s)://hostname:port/conf so fetcher.cpp downloads the
-        //config file to conf, It's an xml file with the parameters of yarn-site.xml, core-site.xml and hdfs.xml.
-        String configCopyCmd = "cp conf " + cfg.getYarnEnvironment().get("YARN_HOME") +
-            "/etc/hadoop/yarn-site.xml";
-
-        //Command to run the executor
-        String executorPathString = myriadExecutorConfiguration.getPath();
```

Comment:
yes, there is some cleanup to be done. I was planning to handle it in a separate PR. Will make a node of all the things to be cleaned up.